From: yangsong8 <yangsong8@xiaomi.com>
Date: Tue, 6 May 2025 21:17:54 +0800
Subject: [PATCH] drivers/usbdev/cdcacm.c: fix check priv->nwrq issue

When enable CONFIG_CDCACM_DISABLE_TXBUF, dev->xmit.buffer always
take one req, so just compare CONFIG_CDCACM_NWRREQS - 1.

Signed-off-by: yangsong8 <yangsong8@xiaomi.com>

diff --git a/drivers/usbdev/cdcacm.c b/drivers/usbdev/cdcacm.c
index 9076547b28..df7d6a53d7 100644
--- a/drivers/usbdev/cdcacm.c
+++ b/drivers/usbdev/cdcacm.c
@@ -1485,7 +1485,12 @@ static void cdcacm_unbind(FAR struct usbdevclass_driver_s *driver,
        */
 
       flags = enter_critical_section();
+
+#ifdef CONFIG_CDCACM_DISABLE_TXBUF
+      DEBUGASSERT(priv->nwrq >= CONFIG_CDCACM_NWRREQS - 1);
+#else
       DEBUGASSERT(priv->nwrq == CONFIG_CDCACM_NWRREQS);
+#endif
 
       while (!sq_empty(&priv->txfree))
         {
